tests/system/test_dbapi.py

```diff
@@ -810,3 +805,33 @@ def test_dml_returning_delete(shared_instance, dbapi_database, autocommit):
     assert cur.fetchone() == (1, "first-name")
     assert cur.rowcount == 1
     conn.commit()
+
+
+def _execute_common_precommit_statements(cursor: Cursor):
```

Comment:
It looks like this code uses pytest?  If so, I would recommend making this a [pytest fixture](https://docs.pytest.org/en/6.2.x/fixture.html#back-to-fixtures).

It looks like this file already has a couple fixtures in it.  I would add another fixture that constructs a connection and cursor; then add this logic as a fixture for initial database data.  The tests above can then depend on both new fixtures; this gets them both a cursor and an initial dataset.

One potential benefit of fixtures is that they support a clean-up phase.  (Just write `yield` in place of `return`, below; then any logic after the `yield` statement is executed as part of fixture tear-down.)  This might, for example, allow you to have a single database that doesn't need to be re-created with every test run (because database creation is expensive), where initial data is populated at the start of the test and any state in the database is cleaned up at the end of the test.

Reply:
This will involve a lot of refactoring so wanted to have a separate PR for it. I have create https://github.com/ankiaga/python-spanner/pull/1 and will raise post merging of this PR. Let me know if this sounds good

Reply:
Sure, sounds good to me.